omap-tutorial/README.md

```diff
@@ -0,0 +1,197 @@
+# OMAP tutorial
+
+This directory contains OMAP exemple for PandaBoardES and Pand5-uevm board.
+
+## OMAP Environment setup:
+
+This part is describing OMAP environement setup for PANDA board.
+
+### Bootloader and Kernel
+This section is describing how to build the different bootloader and kernel components.
+
+- u-boot:
+--------
+For u-boot the upstream mainline is supporting OMAP devices. A simple patch is needed in order to ensure the configuration of the OMAP ABE DPLL to use the SYSCLK for the ABE DPLL.
+
+git clone git://git.denx.de/u-boot.git
+
+Add the #define CONFIG_SYS_OMAP_ABE_SYSCK inside ./include/configs/omap4_panda.h
+```
+make CROSS_COMPILE=arm-none-linux-gnueabi- omap4_panda_config
+make CROSS_COMPILE=arm-none-linux-gnueabi- ARCH=arm
+```
+=> Generation `MLO` and `u-boot.img` for boot partition
+
+- kernel:
+--------
+As OMAP ABE code is not upstream the next tree is sharing a Kernel image based on 3.14 kernel. This image is enabling ABE and still need some improvement for stability.
+
+```
+git clone git://gitorious.org/omap-audio/linux-audio.git
+git checkout origin/sgc/topic/pfw_omap
+make CROSS_COMPILE=arm-none-linux-gnueabi- ARCH=arm omap2plus_defconfig
+make CROSS_COMPILE=arm-none-linux-gnueabi- ARCH=arm uImage -j5
+make CROSS_COMPILE=arm-none-linux-gnueabi- ARCH=arm dtbs
+cd arch/arm/boot
+mkimage -A arm -O linux -T kernel -C none -a 0x80008000 -e 0x80008000 -n "Linux" -d zImage uImage
+```
+=> Copy uImage and `omap4-panda-es.dtb` on the boot partition
+
+
+### OMAP Audio BackEnd Firmware generation
+
+This section is describing how to generated the OMAP4+  ABE FW and the ASoC-DFW header for Audio Firmware binaries.
+
+#### OMAP AESS/ABE FW:
+
+* OMAP ABE tool:
+This tool is used in order to add some FW binary data on top of the firmware. The kernel driver is usem them in order to get all the different FW address for the audio cluster and port configuration.
+
+```
+git clone git://gitorious.org/omap-audio/abefw.git
+./autogen.sh
+./configure --with-linux-dir=/path/to/linux.git
+make
+sudo make install
+./scripts/abe_tool.h 009590
+```
+
+#### ASoC-DFW tool:
+* ASoC-DFW tool
+This tool is used to create all the binary data for the mixer controls, DAPM graph for the Audio driver.
+
+```
+git clone git://gitorious.org/omap-audio/asoc-fw.git
+git checkout origin/sgc/topic/pfw_omap
+./autogen.sh
+./configure --with-linux-dir=/path/to/linux.git --enable-omap4
+make
+sudo make install
+./scripts/abegen.sh
+```
+=> generate `omap4_abe_new` FW file. To be copy inside `/lib/firmware/omap_aess-adfw.bin`
+
+### File System generation
+
+The PFW test has been done with UBUNTU Core File System or Poky minimal image..
+
+- File System:
+-------------
+Download ubuntu core FS.
+=> Untar the FS on the File system partition
+
+* Remove Root password
+sudo gedit /media/rootfs/etc/shadow
```

Comment:
This does not render well. Check the "rich diff" or check here: https://github.com/01org/parameter-framework-samples/blob/sgc/topic/pfw_omap/omap-tutorial/README.md